source/common/upstream/thread_aware_lb_impl.cc

```diff
@@ -30,7 +30,7 @@ void ThreadAwareLoadBalancerBase::refresh() {
   }
 
   {
-    std::unique_lock<std::shared_timed_mutex> lock(factory_->mutex_);
+    absl::WriterMutexLock lock(&factory_->mutex_);
```

Comment:
Note that we had previously wrapped all absl::Mutex and Lock mechanisms in an Envoy-owned abstraction. We might want to continue with that.

Really the main benefit of that is that we have a pure mutex abstraction that had both shared-memory (for hot restart) and local-only implementations, and either could be used in a LockGuard. Maybe that's not useful here because the read/writer lock capability is not in the abstraction.

Reply:
I'd be happy to add the abstraction, but it seems like a lot of extra code for this single use-case. I imagine I'd want to add a SharedLockable interface with a MutexSharedLockable implementation, and WriterLockGuard and ReaderLockGuard classes for RAII-based access. What do you think?

Reply:
Agreed; maybe leave a TODO here that if there's a new usage we should add an abstraction?

And RAII for mutexes is really important in Envoy due to exceptions.

Reply:
Hmm, I don't know that this is the right place for a TODO since it's far away from wherever the new usage would be. I don't know that there's a great place for it, and I'd also rather not have a `TODO(akonradi): maybe do this thing someday` floating around :)